cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js

```diff
@@ -294,10 +304,36 @@ class HydratorPlusPlusTopPanelCtrl {
     seconds = seconds < 10 ? '0' + seconds : seconds;
     minutes = minutes < 10 ? '0' + minutes : minutes;
 
+    this.setDisplayDuration(minutes, seconds);
+  }
+
+  setDisplayDuration(minutes, seconds) {
     this.displayDuration = {
-      minutes: minutes,
-      seconds: seconds
-    };
+      minutes: !minutes ? '--' : minutes,
+      seconds: !seconds ? '--' : seconds,
+    }
+  }
+
+  updateTimerLabelAndTitle(res) {
+    // set default
+    if (!res) {
+      this.timerLabel = 'Duration';
+      this.queueStatus = '';
+      return;
+    }
+
+    const { WAITING, ACQUIRED, INIT, RUNNING } = window.CaskCommon.PREVIEW_STATUS;
+    if (res.status === WAITING) {
+      const runsAheadInQueue = res.positionInWaitingQueue;
+      this.queueStatus = `${runsAheadInQueue} ${runsAheadInQueue === 1? 'run' : 'runs'} ahead in queue`;
```

Comment:
I18n?

Reply:
Good question - we don't have i18n for the angular app (just for the react side).

Reply:
OK then. Is the plan to add I18n as components are converted to React?

Reply:
I believe so. @ajainarayanan might be able to add more context!

Reply:
We don't have a complete story on i18n. We need to rethink how we do i18n. Right now we have a yaml file that we build and ship along with code which takes the away the benefit of i18n (since we can change the language. If we need to we then have to generate a new build). We will have to revisit i18n once we migrate the entirety of angular to react.